src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -507,18 +529,87 @@ void activity_classifier::perform_training_iteration() {
   float average_batch_accuracy = cumulative_batch_accuracy / num_batches;
 
   // Report progress if we have an active table printer.
-  // TODO: Report validation metrics.
-  
-  if (training_table_printer_) {
-    training_table_printer_->print_progress_row(
-        iteration_idx, iteration_idx + 1, average_batch_accuracy, average_batch_loss, progress_time());
+  if (validation_data_iterator_ != nullptr) {
+
+    float cumulative_val_accuracy = 0.f;
+    float cumulative_val_loss = 0.f;
+    size_t val_size = 0;
+
+    validation_data_iterator_->reset();
+
+    while (validation_data_iterator_->has_next_batch()) {
+      data_iterator::batch val =
+          validation_data_iterator_->next_batch(batch_size);
+      std::map<std::string, shared_float_array> results =
+          training_model_->predict({
+              {"input", val.features},
+              {"labels", val.labels},
+              {"weights", val.weights},
+          });
+
+      const shared_float_array &val_output = results.at("output");
+
+      shared_float_array loss = results.at("loss");
+
+      float val_loss = std::accumulate(loss.data(), loss.data() + loss.size(),
+                                       0.f, std::plus<float>());
+
+      cumulative_val_loss += val_loss;
+      float cumulative_per_seq_accuracy = 0.f;
+      for (size_t i = 0; i < val.batch_info.size(); ++i) {
+
+        const shared_float_array &output_chunk = val_output[i];
+        const shared_float_array &label_chunk = val.labels[i];
+        data_iterator::batch::chunk_info info = val.batch_info[i];
+        size_t num_predictions =
+            (info.num_samples + prediction_window - 1) / prediction_window;
+        size_t num_correct_predictions = count_correct_predictions(
+            num_classes, output_chunk, label_chunk, num_predictions);
+
+        cumulative_per_seq_accuracy +=
+            static_cast<float>(num_correct_predictions) / num_predictions;
+      }
+      cumulative_val_accuracy += cumulative_per_seq_accuracy;
```

Comment:
I don't think we need both `cumulative_val_accuracy` and `cumulative_seq_accuracy`. Can just add each sequence's accuracy directly into `cumulative_val_accuracy`